src/app/components/profile/profile.controller.js

```diff
@@ -0,0 +1,91 @@
+angular.module('wisboo').controller('ProfileController', function ($scope) {
+  function getDayClass (data) {
+    const date = data.date;
+    const mode = data.mode;
+    if (mode === 'day') {
+      const dayToCheck = new Date(date).setHours(0, 0, 0, 0);
+      let i;
+      for (i = 0; i < $scope.events.length; i++) {
+        const currentDay = new Date($scope.events[i].date).setHours(0, 0, 0, 0);
+
+        if (dayToCheck === currentDay) {
+          return $scope.events[i].status;
+        }
+      }
+    }
+
+    return '';
+  }
+  // Disable weekend selection
+  function disabled (data) {
+    const date = data.date;
+    const mode = data.mode;
+    return mode === 'day' && (date.getDay() === 0 || date.getDay() === 6);
+  }
+  $scope.today = function () {
+    $scope.dt = new Date();
+  };
+
+  $scope.clear = function () {
```

Comment:
don't use $scope for declaring functions and/or attributes used on the view, use "this" and specify an alias for the controller